gatewayconfig/bluetooth/characteristics/wifi_connect_characteristic.py

```diff
@@ -35,14 +35,9 @@ def check_wifi_status(self):
         nm_state = str(nmcli_custom.device.show('wlan0')['GENERAL.STATE'].split(" ")[0])
 
         # Convert the network manager device state into wifi status response
-        if nm_state == constants.NM_DEVICE_STATE_ACTIVATED:
-            wifi_status = constants.WIFI_CONNECTED
-        elif nm_state == constants.NM_DEVICE_STATE_FAILED:
-            wifi_status = constants.WIFI_ERROR
-        else:
-            wifi_status = constants.WIFI_INVALID_PASSWORD
-
+        wifi_status = constants.WIFI_STATUSES[nm_state]
```

Comment:
Lets wrap this in a try/except KeyError just in case we get an unexpected nm_state for whatever reason.

Reply:
What about using Python dictionary get() method with a default value?

Reply:
What about using Python dictionary get() method with a default value?